assets/src/components/mapPage/mapDisplay.tsx

```diff
@@ -455,13 +455,28 @@ const NearbyStops = ({ stops }: { stops: Stop[] }) => {
   )
   const [nearbyStops, setNearbyStops] = useState<Stop[]>([])
   const map = useMap()
-  map.addEventListener("moveend", () => {
+
+  const filterStopsToBounds = useCallback(() => {
     const bounds = map.getBounds()
     // Only show nearby stations or bus stops
     setNearbyStops(
       stationsAndBus.filter((s) => bounds.contains([s.lat, s.lon]))
     )
-  })
+  }, [map, stationsAndBus, setNearbyStops])
```

Comment:
**question:** trying to wrap my head around this, which of these three dependencies actually changes when you toggle the direction in the RPC?

Reply:
stationAndBus changes, because it's derived from stops